### DISCUSSION TITLE
Porting on a custom STM32 board

### DISCUSSION THREAD

**Discussion by SMarechalBE** [Lora-net/LoRaMac-node#1088] [Q&A]
Hello, 
I'm trying to port the library on a custom STM32L412 board with SX1261 but even with the porting guide, I don't really know where to start. Is there somewhere a step by step explanation of how to port this on an existing project to cubeIDE ? 

Thanks

**Comment by mluis1** [comment]
The NucleoL476 plus the SX1261MBXBAS shield platform is the one closest to your requirements.

You need to create a new directory under boards directory then copy the contents under NucleoL476 directory to your newly created one. Then adapt the individual files according to your MCU and board requirements.
